src/controllers/authLogin.js

```diff
@@ -0,0 +1,40 @@
+/* eslint-disable no-shadow */
+/* eslint-disable no-undef */
+/* eslint-disable no-unused-vars */
+const { join } = require('path');
+const bcrypt = require('bcrypt');
+const { userLoginQuery } = require('../database/queries');
+const { userLoginSchema } = require('../utils/validation');
+const { generateAccessToken } = require('../utils/jwt');
+const CustomError = require('../utils/helpers/customError');
+
+const getLogin = (req, res) => {
+  res.sendFile(join(__dirname, '..', '..', 'public', 'html', 'login.html'));
+};
+
+const authLoginUser = (req, res, next) => {
+  const { email, password } = req.body;
+  const { error, value } = userLoginSchema
+    .validateAsync({ email, password }, { abortEarly: false })
+    .then(() => userLoginQuery(email))
+    .then((data) => {
+      const user = data.rows[0];
```

Comment:
Validate the length of the `data.rows` array before accessing its elements to avoid potential errors. If the array is empty, you can throw an appropriate error or handle it differently based on your application's requirements.